container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,143 @@
+	flag.Var(&ports, "ports", "Augment the ExposedPorts of the previous layer.")
+	flag.Var(&volumes, "volumes", "Augment the Volumes of the previous layer.")
+	flag.Var(&entrypointPrefix, "entrypointPrefix", "Prefix the Entrypoint with the specified arguments.")
+	flag.Var(&env, "env", "Augment the Env of the previous layer.")
+	flag.Var(&command, "command", "Override the Cmd of the previous layer.")
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layerDigestFile, "layerDigestFile", "Layer sha256 hashes that make up this image. The order that these layers are specified matters.")
+	flag.Var(&stampInfoFile, "stampInfoFile", "A list of files from which to read substitutions to make in the provided fields.")
+
+	flag.Parse()
+
+	log.Println("Running the Image Config creator...")
+
+	if *outputConfig == "" {
+		log.Fatalln("Required option -outputConfig was not specified.")
+	}
+
+	configFile := &v1.ConfigFile{}
+	if *baseConfig != "" {
+		configPath, err := ioutil.ReadFile(*baseConfig)
+		if err != nil {
+			log.Fatalf("Failed to read the base image's config file: %v", err)
+		}
+
+		configFile, err = v1.ParseConfigFile(bytes.NewReader(configPath))
+		if err != nil {
+			log.Fatalf("Failed to successfully parse config file json contents: %v", err)
+		}
+	} else {
+		// write out an empty config file.
+		log.Println("baseConfig is empty!")
+	}
+
+	opts := compat.OverrideConfigOpts{
+		ConfigFile:         configFile,
+		OutputConfig:       *outputConfig,
+		CreationTimeString: *creationTimeString,
+		User:               *user,
+		Workdir:            *workdir,
+		NullEntryPoint:     *nullEntryPoint,
+		NullCmd:            *nullCmd,
+		OperatingSystem:    *operatingSystem,
+		CreatedBy:          createdBy,
+		Author:             defaultAuthor,
+		LabelsArray:        labelsArray[:],
+		Ports:              ports[:],
+		Volumes:            volumes[:],
+		EntrypointPrefix:   entrypointPrefix[:],
+		Env:                env[:],
+		Command:            command[:],
+		Entrypoint:         entrypoint[:],
+		Layer:              layerDigestFile[:],
+		StampInfoFile:      stampInfoFile[:],
+	}
+
+	// write out the updated config after overriding config content.
+	if err := compat.OverrideImageConfig(&opts); err != nil {
+		log.Fatalf("Failed to override values in old image config and write to dst %s: %v", err, *outputConfig)
+	}
+
+	log.Printf("Successfully created Image Config at %s.\n", *outputConfig)
+
+	log.Println("Running the Image Manifest creator...")
+
+	var manifestBlob []byte
+	var err error
+	if *baseManifest != "" {
+		manifestBlob, err = ioutil.ReadFile(*baseManifest)
+		if err != nil {
+			log.Fatalf("Failed to read manifest file from %s: %v", *baseManifest, err)
+		}
+	} else {
+		// generating an empty manifest.
+		manifestBlob = []byte("{}")
+	}
+	if err = ioutil.WriteFile(*outputManifest, manifestBlob, os.ModePerm); err != nil {
```

Comment:
actually can't, because we are using manifestBlob outside of if statements, and we need to also error check, so we can't assign to both at once else the compiler says that manifestBlob is unused.